### PR TITLE
Make string capitalisation safe

### PR DESCRIPTION
**Pull request by alxndrsn**
Fix for #435

### CODE REVIEW

File: smssync/src/main/java/org/addhen/smssync/presentation/util/Utility.java [hunk 1]
```diff
@@ -278,6 +278,9 @@ public static String getPhoneNumber(@NonNull Context context, @NonNull PrefsFact
      * @return String
      */
     public static String capitalizeFirstLetter(String text) {
+        if(text == null) return null;
```
Comment (eyedol) [inline smssync/src/main/java/org/addhen/smssync/presentation/util/Utility.java]: Neat but violates coding style. Could you use the standard brace style. We use the standard android [coding style guide](https://source.android.com/source/code-style.html).

Comment (alxndrsn) [inline smssync/src/main/java/org/addhen/smssync/presentation/util/Utility.java]: Can't find the relevant bit of that document; would the following be appropriate?
```java
f(text == null) {
	return null;
} else if(text.length() == 0) {
	return "";
} else if(text.length() == 1) {
	return text.toUpperCase();
} else {
	return text.substring(0, 1).toUpperCase() + text.substring(1);
}
```

Comment (eyedol) [inline smssync/src/main/java/org/addhen/smssync/presentation/util/Utility.java]: Actually your original code holds. However most of the code have braces on the if. This should do.

```java
if (text == null) {
    return null;
}
 
if (text.length() == 0) {
    return "";
}
 
if (text.length() == 1) {
    return text.toUpperCase();
}
return text.substring(0, 1).toUpperCase() + text.substring(1);

```
Thank you.

Comment (alxndrsn) [inline smssync/src/main/java/org/addhen/smssync/presentation/util/Utility.java]: You sure?  Lots of code for a little function!

Comment (eyedol) [inline smssync/src/main/java/org/addhen/smssync/presentation/util/Utility.java]: Heh. Just trying to maintain consistency. In fact it doesn't matter. Make the call.

Comment (alxndrsn) [inline smssync/src/main/java/org/addhen/smssync/presentation/util/Utility.java]: How about that?

**Comment by eyedol** [comment]
Test is breaking

**Comment by eyedol** [comment]
Still [breaking](https://travis-ci.org/ushahidi/SMSSync/builds/124978651). Seems you're missing an import for `assertEquals` and `assertNull`

File: smssync/src/test/java/org/addhen/smssync/presentation/util/UtilityTest.java [hunk 2]
```diff
@@ -38,4 +38,34 @@ public void shouldCheckForIPBasedUrlToBeValid() {
...
+    @Test
+    public void stringCapitalisationShouldWorkForNull() {
+        // expect
+        assertNull(Utility.capitalizeFirstLetter(null));
+    }
+
+    @Test
+    public void stringCapitalisationShouldWorkForEmptyString() {
+        // expect
+        assertEquals("", Utility.capitalizeFirstLetter(""));
```
Comment (eyedol) [inline smssync/src/test/java/org/addhen/smssync/presentation/util/UtilityTest.java]: There's no import for `assertEquals`

File: smssync/src/test/java/org/addhen/smssync/presentation/util/UtilityTest.java [hunk 3]
```diff
@@ -38,4 +38,34 @@ public void shouldCheckForIPBasedUrlToBeValid() {
         boolean actual = Utility.validateUrl("http://192.168.1.4:3000/");
         assertTrue("The provided URL is not a valid one", actual);
     }
+
+    @Test
+    public void stringCapitalisationShouldWorkForNull() {
+        // expect
+        assertNull(Utility.capitalizeFirstLetter(null));
```
Comment (eyedol) [inline smssync/src/test/java/org/addhen/smssync/presentation/util/UtilityTest.java]: There's no import for `assertNull`

**Comment by alxndrsn** [comment]
Fixed :¬)

**Pull request merged**